addons/im_livechat/controllers/main.py

```diff
@@ -70,12 +70,17 @@ def livechat_init(self, channel_id):
     @http.route('/im_livechat/get_session', type="json", auth='public')
     def get_session(self, channel_id, anonymous_name, **kwargs):
         # if geoip, add the country name to the anonymous name
+        country_id = None
         if request.session.geoip:
-            anonymous_name = anonymous_name + " ("+request.session.geoip.get('country_name', "")+")"
+            # get the country of the anonymous person, if any
+            country_code = request.session.get('country_code', "")
+            country_id = request.env['res.country'].sudo().search([('code', '=', country_code)], limit=1).id if country_code else None
+            anonymous_name = anonymous_name + " (" + request.session.get('country_name', "") + ")"
```

Comment:
While we are at it, can you change it into `_("%s (%s))" % (anonymous_name, coutnry_code) if country code else anonymous_name`. So it will be translatable.